### ISSUE TITLE
"Return-Path" setting in emails fired by WireMail

### ISSUE THREAD

**Issue by processwired** [ryancramerdesign/ProcessWire#1421]
I'm having delivery issues from WireMail because it doesn't set the "Return-path" header setting explicitly, as far as I know.

For instance, the "forgot password" module emails don't get delivered, because of reasons like the following (excerpt from mail log): "550 permanent failure for one or more recipients (myemail@myserver.com:550 rejecting spoofed message) (in reply to end of DATA command))"

Basically, the others party mail server is of the opinion that the mail is spoofed because "Return-path" setting and "From" setting are not identical. The "Return-path" is set by the local server, configured in generic php.ini, the "From" setting is configured by "forgot password" module.

 Any means to cure that? I guess it might help to set "Return-path" explicitly to the same value the "From" setting has.

Another minor problem: german translation of subject line of "forgot password" emails makes use of umlauts. Amavis is complaining about non encoded 8-bit data then: "X-Amavis-Alert: BAD HEADER SECTION, Non-encoded 8-bit data (char C3 hex): Subject: Informationen zur Zur\303\203\302\274cksetzung des [...]"

**Comment by processwired** [comment]
UPDATE: in WireMail.php it asks for a config variable called "phpMailAdditionalParameters". Solution for first problem seems to be: add `$config->phpMailAdditionalParameters = '-f mymail@myserver.com';`in /site/config.php. Afterwards, the "Return-Path" header is set to <mymail@myserver.com>.

**Comment by ryancramerdesign** [comment]
As far as I know that Return-Path has to be set by the '-f' parameter when using PHP mail(), for situations where the server configuration needs it. I haven't had to use it before, but I know others have. I can't think of anything we can change here, short of the phpMailAdditionalParameters option already provided, but let me know if you can think of anything.

Regarding the subject line for forgot password emails, any idea what (if anything) we should change here? I've not heard of the issue occurring with anyone else thus far, so wondering if it might be server specific?

**Comment by ryancramerdesign** [comment]
Btw, web servers aren't always the idea platform for sending mail (depending on the host). In a lot of situations its better to install either the WireMailSwiftMailer or WireMailSMTP module so that the email sending can be delegated to an outside SMTP server like Mandrill, Gmail, or your POP3 server.

**Comment by cb2004** [comment]
Absolutely agree with what Ryan says.

**Comment by processwired** [comment]
Thanks a lot, Ryan, for the useful hints. I experimented with WireMailSwiftMailer and WireMailSMTP modules. WireMailSMTP is kind of more "matured" than WireMailSwiftMailer, and the problem described is the same with WireMailSwiftMailer, so at the end I went with WireMailSMTP (though I like the base of WireMailSwiftMailer - SwiftMailer - on it's own pretty much, it never let me down for years and it is actively developed). I think these two modules are a good way to enhance the basic functionality of what ProcessWire offers in this regard. At least these two modules are a good source for an individual custom module, if needed. All in all: I'm going to close this issue.
